ngmix/admom.py

```diff
@@ -32,54 +72,51 @@ class Admom(object):
         Largest allowed shift in the centroid, relative to
         the initial guess.  Default 5.0 (5 pixels if the jacobian
         scale is 1)
+    rng: numpy.random.RandomState
+        Random state for creating full gaussian guesses based
+        on a T guess
     """
 
-    def __init__(self, obs, maxiter=200, shiftmax=5.0,
-                 etol=1.0e-5, Ttol=0.001,  # noqa
+    def __init__(self,
+                 maxiter=DEFAULT_MAXITER,
+                 shiftmax=DEFAULT_SHIFTMAX,
+                 etol=DEFAULT_ETOL,
+                 Ttol=DEFAULT_TTOL,
                  rng=None,
                  **unused_keys):
 
-        if isinstance(obs, ObsList):
-            if len(obs) != 1:
-                raise ValueError("at most one epoch can be fit")
-            obs = obs[0]
-        elif isinstance(obs, MultiBandObsList):
-            if len(obs) > 1:
-                raise ValueError("at most one band can be fit")
-            else:
-                obs = obs[0]
-                if len(obs) != 1:
-                    raise ValueError("at most one epoch can be fit")
-                obs = obs[0]
-
-        self._obs = obs
         self._set_conf(maxiter, shiftmax, etol, Ttol)
 
         self.rng = rng
 
-    def go(self, guess):
+    def go(self, obs, guess):
         """
         run the adpative moments
 
         parameters
         ----------
-        guess_gmix: ngmix.GMix or a float
-            A guess for the fitter.  Can be a full gaussian
-            mixture or a single value for T, in which case
-            the rest of the parameters are random numbers
-            about the jacobian center and zero ellipticity
+        obs: Observation
+            ngmix.Observation
+        guess: ngmix.GMix or a float
+            A guess for the fitter.  Can be a full gaussian mixture or a single
+            value for T, in which case the rest of the parameters for the
+            gaussian are generated.
         """
         from .admom_nb import admom
 
-        guess_gmix = self._get_guess(guess)
+        if not isinstance(obs, Observation):
+            raise ValueError("input obs must be an Observation")
+        self._obs = obs
```

Comment:
Should we be caching this? Maybe the right thing here are FitResult objects that can do things like make the mixture and get the parameters etc. This way the fitters are not caching info about the data they fit. 

I am not 100% sure I like that design, but it feels odd to have the fitter not have the obs in the constructor but then cache it later.

Reply:
I'm going to punt on this